### PR TITLE
[FN] Introduce variable thresholds for BIP9

### PR DESCRIPTION
**Pull request by zeptin**

### CODE REVIEW

**Comment by fassadlr** [comment]
@zeptin  can you fix the conflicts?

**Comment by zeptin** [comment]
@fassadlr done

File: src/Stratis.Bitcoin.IntegrationTests/ColdStakingTests.cs [hunk 1]
```diff
@@ -37,14 +37,12 @@ public void ColdStakingActivatedOnStratisNode()
                 var network = new StratisOverrideRegTest();
 
                 // Set the date ranges such that ColdStaking will 'Start' immediately after the initial confirmation window.
-                network.Consensus.BIP9Deployments[StratisBIP9Deployments.ColdStaking] = new BIP9DeploymentsParameters("Test", 1, 0, DateTime.Now.AddDays(50).ToUnixTimestamp());
+                // Also reduce the minimum number of 'votes' required within the conformation window to reach 'LockedIn' state.
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.IntegrationTests/ColdStakingTests.cs]:
```suggestion
                // Also reduce the minimum number of 'votes' required within the confirmation window to reach 'LockedIn' state.
```

**Comment by dangershony** [comment]
@fassadlr I agree, I just argue it can be more flexible, right now we only have the ability to define a threshold per deployment we could also add the window time to the mix with probably not much more work.

But this is a good start indeed.

**Pull request merged**